sw/device/boot_rom/boot_rom.c

```diff
@@ -47,12 +48,6 @@ void _boot_start(void) {
   // return.
   _flash_header.entry();
 
-  // If there's a stop address, write to it.
-  if (kDeviceStopAddress != 0) {
-    mmio_region_t end_sim_addr = mmio_region_from_addr(kDeviceStopAddress);
-    // We write `0xFFFFFFFE` to differentiate from `abort` (which writes
-    // `0xFFFFFFFF`), and the `main` function in the flash image (which is
-    // likely to return `0x0` or `0x1` to match unix exit codes).
-    mmio_region_write32(end_sim_addr, 0x0, 0xFFFFFFFE);
-  }
+  // If the flash image returns, we should abort anyway.
+  abort();
```

Comment:
Please invoke `test_status_set(kTestStatusFailed)` instead.

Reply:
We're trying to avoid having non-test code in the main rom images (though this matters less for boot_rom now), i think I'm just going to leave it as a call to `abort`, as we shouldn't hit this in the tests we have.

Reply:
It doesn't get optimized out yet, sadly, but we're working on ensuring it does.